lib/cocoapods/installer/analyzer.rb

```diff
@@ -226,6 +226,57 @@ def update_repositories
 
       private
 
+      # Copies the pod_targets of any of the app extension aggregate targets into
+      # their potential host aggregate target, if that potential host aggregate target's
+      # user_target hosts any of the app extension aggregate targets' user_targets
+      #
+      # @param  [AggregateTarget] aggregate_target the aggregate target whose user_target
+      #         might host one or more of the app extension aggregate targets' user_targets
+      #
+      # @param  [Array<AggregateTarget>] extension_aggregate_targets the aggregate targets
+      #         representing the app extension targets to be integrated
+      #
+      def copy_extension_pod_targets_to_host(aggregate_target, extension_aggregate_targets)
+        return if aggregate_target.requires_host_target?
+        # Get the uuids of the aggregate_target's user_target's extension targets if any
+        extension_uuids = aggregate_target.user_project.extensions_for_native_target(aggregate_target.user_targets[0]).map(&:uuid)
+        return if extension_uuids.empty?
+        extension_aggregate_targets.each do |extension_target|
+          next unless extension_uuids.include? extension_target.user_targets[0].uuid
+          raise "#{aggregate_target.name} must use_frameworks! because it is hosting an extension." unless aggregate_target.requires_frameworks?
```

Comment:
Oh I rewrote this a few times, and the most recent time I didn't finish the sentence 😫. I meant it to look like this:
 `"#{aggregate_target.name} must call use_frameworks! because it is hosting an extension that calls use_frameworks!."`

Would you agree that this is accurate? Or is it okay that a target that doesn't use frameworks ends up embedding a framework?